### PR TITLE
feat: Recovery tracking

### LINKED ISSUE #2835: [Recovery] Tracking
## What is the feature about

Adding tracking to the new recovery feature implementation:

- Dashboard widgets and interaction
- Recovery queue and interaction
- Recovery settings and interaction
- Email settings and interaction

## The list of requirements

- [ ] Align with the relevant parties regarding what needs tracking.
- [ ] Implement aforementioned tracking.

## Links

- [Feature spec.](https://www.notion.so/safe-global/Recovery-feature-spec-d873c33a27c24cfabc91c85dbcf3ea35)

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #2835 

## How this PR fixes it

- Adds recovery events and documents them ([Notion](https://www.notion.so/safe-global/Google-Analytics-custom-events-bfd072d2bb704394860a59724dc3c66f?pvs=4#5f9ae8c0a430458895cee38e5622f43f))
- Extends the `useTxStepper` hook to receive an optional argument for `eventCategory` and track Next/Back clicks

## ToDos

- [x] The `recovery_attempt` event is sent before the user submits the transaction in their wallet
- [x] No event is sent if the expired recovery tx is canceled

## How to test it

Open a Safe, go through the list of events and make sure they are sent as described

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/recovery/CancelRecoveryButton/index.tsx [hunk 1]
```diff
@@ -37,9 +39,11 @@ export function CancelRecoveryButton({
     e.preventDefault()
 
     if (isOwner) {
+      trackEvent(RECOVERY_EVENTS.CANCEL_RECOVERY)
```
Comment (katspaugh) [inline src/components/recovery/CancelRecoveryButton/index.tsx:42]: Why behind an if? A click is a click.

File: src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx [hunk 2]
```diff
@@ -41,7 +48,11 @@ export function _RecoveryProposalCard({ orientation = 'vertical', onClose, safe,
   } regain access by updating the owner list.`
 
   const link = (
-    <ExternalLink href={HelpCenterArticle.RECOVERY} title="Learn more about the Account recovery process">
+    <ExternalLink
+      href={HelpCenterArticle.RECOVERY}
+      onClick={onLearnMoreClick}
```
Comment (katspaugh) [inline src/components/recovery/RecoveryCards/RecoveryProposalCard.tsx:53]: Track component migth work as well.

File: src/components/tx-flow/useTxStepper.tsx [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
-  const nextStep = useCallback((entireData: T) => {
-    setData(entireData)
-    setStep((prevStep) => prevStep + 1)
-  }, [])
+  const nextStep = useCallback(
+    (entireData: T) => {
+      setData(entireData)
+      setStep((prevStep) => {
+        if (eventCategory) {
+          trackEvent({ action: 'Next', category: eventCategory, label: prevStep })
```
Comment (katspaugh) [inline src/components/tx-flow/useTxStepper.tsx:13]: What's this event?
* `Next`/`Back` is not specific enough. It should be something like `Tx flow next`.
* It should be coming from an enum in `analytics/events`.

File: src/services/analytics/events/recovery.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  CANCEL_RECOVERY: {
+    action: 'Cancel recovery',
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  SHOW_ADVANCED: {
+    action: 'Show advanced recovery settings',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:27]: Too long.

File: src/services/analytics/events/recovery.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  LEARN_MORE: {
+    action: 'Learn more',
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  GO_BACK: {
+    action: 'Go back on cancel',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:42]: The other evenrs in this file have "recovery" in the name, but these don't and they can be ambigous unless the category is used as well.

File: src/services/analytics/events/transactions.ts [hunk 6]
```diff
@@ -26,6 +26,13 @@ export enum TX_TYPES {
   typed_message = 'typed_message',
   safeapps = 'safeapps',
   walletconnect = 'walletconnect',
+
+  // Recovery
+  recovery_setup = 'recovery_setup',
+  recovery_edit = 'recovery_edit',
+  recovery_remove = 'recovery_remove',
+  recovery_init = 'recovery_init',
```
Comment (katspaugh) [inline src/services/analytics/events/transactions.ts:34]: I would rename this to `recovery_attempt`.

File: src/components/recovery/CancelRecoveryButton/index.tsx (hunk 1 above)
Comment (usame-algan) [reply src/components/recovery/CancelRecoveryButton/index.tsx:42]: It is also possible for non-owners to cancel the recovery tx after it has expired. Not sure if we should track that as a separate event or if both should be counted under the same event. cc @alicecha

File: src/services/analytics/events/recovery.ts (hunk 5 above)
Comment (usame-algan) [reply src/services/analytics/events/recovery.ts:42]: We can make these actions longer and say `Learn more about recovery` etc.

File: src/components/recovery/CancelRecoveryButton/index.tsx (hunk 1 above)
Comment (usame-algan) [reply src/components/recovery/CancelRecoveryButton/index.tsx]: I've changed it for now so we track this event in both cases.

File: src/services/analytics/events/recovery.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+export const SETUP_RECOVERY_CATEGORY = 'setup-recovery'
+export const START_RECOVERY_CATEGORY = 'start-recovery'
+export const REMOVE_RECOVERY_CATEGORY = 'remove-recovery'
+export const CANCEL_RECOVERY_CATEGORY = 'cancel-recovery'
+
+const RECOVERY_CATEGORY = 'recovery'
+
+export const RECOVERY_EVENTS = {
+  SETUP_RECOVERY: {
+    action: 'Set up recovery',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:13]: `Enable recovery`?

File: src/services/analytics/events/recovery.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  RECOVERY_SETTINGS: {
+    action: 'Recovery settings',
+    category: RECOVERY_CATEGORY,
+    event: EventType.META,
+  },
+  EDIT_RECOVERY: {
+    action: 'Edit recovery setup',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:23]: `Edit recovery`

File: src/services/analytics/events/recovery.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  DISMISS_PROPOSAL_CARD: {
+    action: 'Dismiss recovery proposal card',
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  LEARN_MORE: {
+    action: 'Learn more about recovery',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:53]: `Recovery info click`

File: src/services/analytics/events/recovery.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    category: RECOVERY_CATEGORY,
+    event: EventType.CLICK,
+  },
+  GO_BACK: {
+    action: 'Go back on cancel recovery',
```
Comment (katspaugh) [inline src/services/analytics/events/recovery.ts:58]: `Recovery cancellation back`

**Review by katspaugh — COMMENTED** [review]
Some suggestions for more concise event actions

File: src/services/analytics/events/recovery.ts (hunk 7 above)
Comment (usame-algan) [reply src/services/analytics/events/recovery.ts:13]: Current suggestion is to update it to "Start recovery set-up" once the new modal is implemented so I would leave it as it is for now.

**Review by katspaugh — APPROVED** [review]
Gut diese 👍

**Comment by usame-algan** [comment]
The remaining events will be added with #2930

**Pull request merged**